libanki/src/main/java/com/ichi2/anki/libanki/Collection.kt

```diff
@@ -478,44 +482,204 @@ class Collection(
 
+    /** Save card changes to database. */
+    @LibAnkiAlias("update_cards")
     fun updateCards(
         cards: Iterable<Card>,
         skipUndoEntry: Boolean = false,
     ): OpChanges = backend.updateCards(cards.map { it.toBackendCard() }, skipUndoEntry)
 
+    /** Save card changes to database. */
+    @LibAnkiAlias("update_card")
     fun updateCard(
         card: Card,
         skipUndoEntry: Boolean = false,
     ): OpChanges = updateCards(listOf(card), skipUndoEntry)
 
+    @LibAnkiAlias("get_note")
+    @CheckResult
     fun getNote(id: NoteId): Note = Note(this, id)
 
+    /** Save note changes to database. */
+    @LibAnkiAlias("update_notes")
+    fun updateNotes(notes: Iterable<Note>): OpChanges =
+        backend.updateNotes(
+            notes = notes.map { it.toBackendNote() },
+            skipUndoEntry = false,
+        )
+
     /**
-     * Notes ******************************************************************** ***************************
+     * Save note changes to database.
      */
-    fun noteCount(): Int = db.queryScalar("SELECT count() FROM notes")
+    @CheckResult
+    @LibAnkiAlias("update_note")
+    fun updateNote(note: Note): OpChanges = backend.updateNotes(notes = listOf(note.toBackendNote()), skipUndoEntry = false)
 
-    /**
-     * Return a new note with the model derived from the deck or the configuration
-     * @param forDeck When true it uses the model specified in the deck (mid), otherwise it uses the model specified in
-     * the configuration (curModel)
-     * @return The new note
+    /*
+     * Utils
+     * ***********************************************************
+     */
+
+    @LibAnkiAlias("nextID")
+    @RustCleanup("Python returns 'Any' - may fail for Double?")
+    fun nextId(
+        type: String,
+        inc: Boolean = true,
+    ): Long {
+        val capitalized = type.replaceFirstChar { if (it.isLowerCase()) it.titlecase(Locale.getDefault()) else it.toString() }
+        val type = "next$type"
+        val id = config.get<Long>(type, 1)!!
+        if (inc) {
+            config.set(type, id + 1)
+        }
+        return id
+    }
+
+    /*
+     * Notes
+     * ***********************************************************
      */
-    fun newNote(forDeck: Boolean = true): Note = newNote(notetypes.current(forDeck))
 
     /**
      * Return a new note with a specific model
      * @param notetype The model to use for the new note
      * @return The new note
      */
+    @LibAnkiAlias("new_note")
     fun newNote(notetype: NotetypeJson): Note = Note.fromNotetypeId(this, notetype.id)
 
+    @LibAnkiAlias("add_note")
+    fun addNote(
+        note: Note,
+        deckId: DeckId,
+    ): OpChanges {
+        val out = backend.addNote(note.toBackendNote(), deckId)
+        note.id = out.noteId
+        return out.changes
+    }
+
+    @LibAnkiAlias("add_notes")
+    @RustCleanup("Implement")
+    @Deprecated("Needs implementation")
```

Comment:
Your choice, but using Deprecated doesn't feel right. The crash on TODO() is going to make it obvious that the method requires attention.

```suggestion
    @RustCleanup("Needs implementation")
```

Reply:
I've changed this to the following. This makes it obvious that it's not callable

`WARNING` shows it up as failing in the IDE, which is also useful as a signal that something is wrong

```kotlin
    @Deprecated("Needs implementation", level = DeprecationLevel.HIDDEN)
```